### PR TITLE
Add unicode.Rune.isNumeric, isDigit, isDecimal

### PR DESCRIPTION
**Pull request by nxshock**

### CODE REVIEW

**Comment by data-man** [comment]
In your PR, only ASCII digits are added. It isn't correct.

File: lib/pure/unicode.nim [hunk 1]
```diff
@@ -1357,6 +1418,11 @@ proc isUpper*(c: Rune): bool {.rtl, extern: "nuc$1", procvar.} =
   if p >= 0 and c == tolowerSinglets[p]:
     return true
 
+proc isDigit*(c: Rune): bool {.rtl, extern: "nuc$1", procvar.} =
+  ## Returns true iff ``c`` is a digit character
+  var c = RuneImpl(c)
+  return c in digitRanges
```
Comment (nxshock) [inline lib/pure/unicode.nim]: Looks like I can't check for digit like this... I need to rewrite this line.

**Comment by data-man** [comment]
@nxshock

Which Unicode version you used?
Version 10.0 contains 590 characters (in Nd category).

See also [nim-unicodedb](https://github.com/nitely/nim-unicodedb) and [nim-unicodeplus](https://github.com/nitely/nim-unicodeplus) libraries.
nim-unicodeplus contains these procs:
```
isDecimal
isDigit
isNumeric
```
Just for compatibility reasons.

**Comment by nxshock** [comment]
`digitRanges` contains 590 characters (59 groups * 10 chars) from Unicode 10.

What kind of compatibility you are mean? As described, nim-unicodeplus _is not drop-in replacement_, so it must be used like this:
```nim
import unicode except isDigit # like README.md example
import unicodeplus
```

I think if we have Rune.isAlpha, we need to have Rune.isDigit in std lib. :)

**Comment by data-man** [comment]
@nxshock

I mean that it would be nice to have ```isDecimal``` and ```isNumeric```. :)

**Comment by nxshock** [comment]
Unfortunately I'm neither Unicode guru not English guru.

As described [here](https://www.unicode.org/Public/10.0.0/ucd/extracted/DerivedNumericType.txt), there are:
* 865 numeric chars
* 128 digit chars
* 590 decimal chars

Should we use same naming rules or not? If yes, I just implemented `Rune.isDecimal`, not `Rune.isDigit`...

**Comment by data-man** [comment]
@nxshock

You implemented checking for a decimal digits.
So ```isDecimalDigit``` will be better name, IMO.

LGTM. Let's wait for other opinions.
> nor English guru

Me too :).

Some useful links:
[Category 'Number, Decimal Digit'](http://www.fileformat.info/info/unicode/category/Nd/index.htm)
[Category 'Number, Letter'](http://www.fileformat.info/info/unicode/category/Nl/index.htm)
[Category 'Number, Other'](http://www.fileformat.info/info/unicode/category/No/index.htm)

**Comment by Varriount** [comment]
The only thing I have to add to the table is that it looks like all those new procedures could share a common template implementation.

**Comment by Araq** [comment]
Nice job, but all of these tables need to be generated from the Unicode spec. Otherwise it's a maintenance burden. I know the current code does it this way, but I don't want to add more of the same bad stuff.

**Comment by Yardanico** [comment]
Like this library -  https://github.com/nitely/nim-unicodedb ? It generates all info from UCD

**Comment by Araq** [comment]
@Yardanico Probably.

**Comment by Araq** [comment]
No feedback, closing.

**Pull request closed without merging**